.github/workflows/push.yml

```diff
@@ -0,0 +1,27 @@
+name: actions
+
+on: push
+
+jobs:
+  # https://github.com/pre-commit/action
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions/setup-python@v1
+      - name: set PY
+        run: echo "::set-env name=PY::$(python --version --version | sha256sum | cut -d' ' -f1)"
+      - uses: actions/cache@v1
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.PY }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - uses: pre-commit/action@v1.0.1
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build the stack
+        run: docker-compose build
+      - name: Test
+        run: docker-compose run --rm web bash -c "flake8 . && python wait_for_postgres.py && ./manage.py test"
```

Comment:
My impression is that github actions might benefit from having a separate workflow for something like linting and testing. Your linting workflow could be largely derived by the one from the marketplace that I linked. I feel like that's the main value - you can break up large CI processes into small tasks that run on specific triggers. Some might share triggers.